### PR TITLE
622 newly defined units properly translate when importing from sbml files

### LINKED ISSUE #622: Newly defined units should be properly translated when importing from sbml files
should be calculated as multiplier times 10 to the power of scale and the exponent should only means direct or inverse unit

### PR DESCRIPTION
**Pull request by abdelr**
Fixes #622 (Using scale properly when importing unit definitions from SBML Files)

### CODE REVIEW

File: src/MoBi.Engine/Sbml/UnitDefinitionImporter.cs [hunk 1]
```diff
@@ -31,16 +49,19 @@ public class UnitDefinitionImporter : SBMLImporter, IStartable, IUnitDefinitionI
       private readonly IMoBiDimensionFactory _moBiDimensionFactory;
       private readonly IDictionary<int, Unit> _baseUnitsDictionary;
       private readonly IDictionary<string, UnitConvertionInfo> _unitConvertionDictionary;
+      private IDimensionFactory _dimensionFactory;
+      public IReadOnlyDictionary<string, IDimension> ConvertionDictionary { get => _unitConvertionDictionary.ToDictionary(kv => kv.Key, kv => kv.Value.Dimension); }
       private IDictionary<string, string> _sbmlUnitsSynonyms = new Dictionary<string, string>()
```
Comment (msevestre) [inline src/MoBi.Engine/Sbml/UnitDefinitionImporter.cs:54]: @abdelr Try to think Cache over Dictionary . cache exposes just so many more convenience methods
for instance you can remove the methods where you check if the value is in the cache completely by defining the behavior at the cache level

```
_sbmlUnitsSynonyms  = new Cache<string, string>(onMissingkeys = k=>k)
```
e.g what happens when the key is missing. we return the key.

**Review by msevestre — COMMENTED** [review]
Merging but let's try to avoid using Dictionary unless absolutely required please

File: src/MoBi.Engine/Sbml/UnitDefinitionImporter.cs (hunk 1 above)
Comment (abdelr) [reply src/MoBi.Engine/Sbml/UnitDefinitionImporter.cs:54]: Check.

**Comment by msevestre** [comment]
@abdelr So many merge conflictts that I resolved by hand... Going to merge if the PR builds But can you check that this is all correct

**Comment by msevestre** [comment]
the diff is still there in the PR
Otherwise yes, check the content of the file UnitDefinitionImporter.cs: and make sure it is correct

**Pull request merged**